### PR TITLE
remove deprecated methods by not stubbing value objects

### PR DESCRIPTION
**Pull request by elharo**
@mabrarov

### CODE REVIEW

File: src/test/java/org/apache/maven/plugins/ear/AbstractEarTestBase.java [hunk 1]
```diff
@@ -33,88 +35,42 @@
...
-            String artifactId = artifactsId[i];
-            String type = getData( types, i, DEFAULT_TYPE );
-            String groupId = getData( groupsId, i, DEFAULT_GROUPID );
-            String classifier = getData( classifiers, i, null );
-            result.add( new ArtifactTestStub( groupId, artifactId, type, classifier ) );
+            String artifactId = artifactIds[i];
+            String classifier = classifiers == null ? null : classifiers[i];
+            Artifact artifactTestStub = new DefaultArtifact(
+                DEFAULT_GROUPID, artifactId, "1.0", "compile", DEFAULT_TYPE, classifier, artifactHandler );
+
```
Comment (mabrarov) [inline src/test/java/org/apache/maven/plugins/ear/AbstractEarTestBase.java:60]: It looks like there are trailing spaces in this line.

File: src/test/java/org/apache/maven/plugins/ear/AbstractEarTestBase.java [hunk 2]
```diff
@@ -33,88 +35,42 @@
 
     public static final String DEFAULT_GROUPID = "eartest";
```
Comment (mabrarov) [inline src/test/java/org/apache/maven/plugins/ear/AbstractEarTestBase.java:36]: Could be:

```java
    protected static final String DEFAULT_GROUP_ID = "eartest";
```

like `org.apache.maven.plugins.ear.util.ArtifactRepositoryTest#MAIN_ARTIFACT_ID`

Comment (elharo) [reply src/test/java/org/apache/maven/plugins/ear/AbstractEarTestBase.java]: done

File: src/test/java/org/apache/maven/plugins/ear/AbstractEarTestBase.java (hunk 1 above)
Comment (elharo) [reply src/test/java/org/apache/maven/plugins/ear/AbstractEarTestBase.java]: done

**Review by elharo — COMMENTED** [review]
PTAL

**Comment by mabrarov** [comment]
I'll need next 24-48 hrs to complete with review of this PR. I haven't forgotten about it :) Just busy with my job duties.

File: src/test/java/org/apache/maven/plugins/ear/util/ArtifactRepositoryTest.java [hunk 3]
```diff
@@ -34,14 +36,15 @@
...
+
+    private ArtifactTypeMappingService artifactTypeMappingService = new ArtifactTypeMappingService();
 
     @Test
     public void testEmptyRepository()
     {
-        ArtifactTypeMappingService artifactTypeMappingService = new ArtifactTypeMappingService();
-        ArtifactRepository repo =
-            new ArtifactRepository( createArtifacts( null ), MAIN_ARTIFACT_ID, artifactTypeMappingService );
+        Set<Artifact> artifacts = new HashSet<>();
```
Comment (mabrarov) [inline src/test/java/org/apache/maven/plugins/ear/util/ArtifactRepositoryTest.java:46]: Won't `Collections#emptySet` be more readable? Like:

```java
        ArtifactRepository repo =
            new ArtifactRepository( Collections.<Artifact>emptySet(), MAIN_ARTIFACT_ID, artifactTypeMappingService );
```

**Review by mabrarov — APPROVED** [review]

File: src/test/java/org/apache/maven/plugins/ear/util/ArtifactRepositoryTest.java (hunk 3 above)
Comment (elharo) [reply src/test/java/org/apache/maven/plugins/ear/util/ArtifactRepositoryTest.java:46]: IMHO nested functions are far less readable. There's no name to explain what the argument is. The line is longer and more likely to run off the edge of the screen or have to be wrapped. And it's definitely more trouble to work with in a debugger. Short, simple lines that do exactly one thing are my preference.

**Pull request merged**